homeassistant/components/skybell/binary_sensor.py

```diff
@@ -12,90 +12,94 @@
     BinarySensorEntity,
     BinarySensorEntityDescription,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ENTITY_NAMESPACE, CONF_MONITORED_CONDITIONS
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-from . import DEFAULT_ENTITY_NAMESPACE, DOMAIN as SKYBELL_DOMAIN, SkybellDevice
+from . import DOMAIN, SkybellEntity
+from .const import DATA_COORDINATOR, DATA_DEVICES
 
 SCAN_INTERVAL = timedelta(seconds=10)
 
 
-BINARY_SENSOR_TYPES: dict[str, BinarySensorEntityDescription] = {
-    "button": BinarySensorEntityDescription(
-        key="device:sensor:button",
+BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key="button",
         name="Button",
         device_class=BinarySensorDeviceClass.OCCUPANCY,
     ),
-    "motion": BinarySensorEntityDescription(
-        key="device:sensor:motion",
+    BinarySensorEntityDescription(
+        key="motion",
         name="Motion",
         device_class=BinarySensorDeviceClass.MOTION,
     ),
-}
-
+)
 
+# Deprecated in Home Assistant 2022.3
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(
-            CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE
-        ): cv.string,
+        vol.Optional(CONF_ENTITY_NAMESPACE, default=DOMAIN): cv.string,
         vol.Required(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
             cv.ensure_list, [vol.In(BINARY_SENSOR_TYPES)]
         ),
     }
 )
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the platform for a Skybell device."""
-    skybell = hass.data[SKYBELL_DOMAIN]
+    """Set up Skybell switch."""
+    skybell = hass.data[DOMAIN][entry.entry_id]
 
-    binary_sensors = [
-        SkybellBinarySensor(device, BINARY_SENSOR_TYPES[sensor_type])
-        for device in skybell.get_devices()
-        for sensor_type in config[CONF_MONITORED_CONDITIONS]
-    ]
+    sensors = []
+    for sensor in BINARY_SENSOR_TYPES:
+        for device in skybell[DATA_DEVICES]:
+            sensors.append(
+                SkybellBinarySensor(
+                    skybell[DATA_COORDINATOR],
+                    device,
+                    sensor,
+                    entry.entry_id,
+                )
+            )
 
-    add_entities(binary_sensors, True)
+    async_add_entities(sensors, True)
```

Comment:
```suggestion
    async_add_entities(sensors)
```

Looks like these are coordinated entities now so we probably don't need update_before_add=True